sources/csv/data.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csv
+
+import (
+	csvReader "encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/civil"
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+type Column struct {
+	Column_name string `json:"column_name"`
+	Type_name   string `json:"type_name"`
+}
+
+// Harbourbridge accepts a manifest file in the form of a json which unmarshalls into the Table struct.
+type Table struct {
+	Table_name    string   `json:"table_name"`
+	File_patterns []string `json:"file_patterns"`
```

Comment:
What are the scenarios where we would have multiple file patterns per table ?
I'm guessing this is to be compatible with the existing structure, but good to know if there is a valid use case

Reply:
This I did mostly for consistency reasons. I think most poc dbs will just have 1 csv file per table. But if some user has a table split over multiple files they can use this functionality. Thought I'm not sure how frequently that occurs.